test/e2e/framework/addon/vault/vault.go

```diff
@@ -105,44 +106,136 @@ func (v *Vault) Setup(cfg *config.Config) error {
 	}
 	v.details.Kubectl = cfg.Kubectl
 	v.chart = &chart.Chart{
-		Base:        v.Base,
-		ReleaseName: "chart-vault-" + v.Name,
-		Namespace:   v.Namespace,
-		ChartName:   cfg.RepoRoot + "/test/e2e/charts/vault",
-		// doesn't matter when installing from disk
-		ChartVersion: "0",
+		Base:         v.Base,
+		ReleaseName:  "chart-vault-" + v.Name,
+		Namespace:    v.Namespace,
+		ChartName:    "hashicorp/vault",
+		ChartVersion: "0.22.0",
+		Repo: chart.Repo{
+			Name: "hashicorp",
+			Url:  "https://helm.releases.hashicorp.com",
+		},
 		Vars: []chart.StringTuple{
 			{
-				Key:   "vault.publicKey",
-				Value: base64.StdEncoding.EncodeToString(v.details.VaultCert),
+				Key:   "global.tlsDisable",
+				Value: "false",
+			},
+			{
+				Key: "server.standalone.config",
+				Value: `
+				listener "tcp" {
+							address = "[::]:8200"
+							cluster_address= "[::]:8201"
+							tls_disable = false
+							tls_cert_file = "/vault/tls/server.crt"
+							tls_key_file = "/vault/tls/server.key"
+				}`,
+			},
+			{
+				Key:   "server.extraArgs",
+				Value: "-dev -dev-listen-address=[::]:8202",
+			},
+			{
+				Key:   "server.extraEnvironmentVars.VAULT_DEV_ROOT_TOKEN_ID",
+				Value: "vault-root-token",
```

Comment:
can we use the helm chart `server.dev` options here instead of the flag & environment var?

Reply:
As far as I understand the helm chart: As soon as you enable `server.dev` you cannot specify a config file anymore.
And without a config file it is hard to enable TLS.

Reply:
Thank you for explaining, this very counter-intuitive.